src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigInstallTask.java

```diff
@@ -92,6 +92,9 @@ public void execute(final InstallationContext ctx) {
                 }
 
                 Configuration config = ConfigUtil.getConfiguration(this.getConfigurationAdmin(), this.factoryPid, this.configPid);
+                if (config == null) {
+                    config = ConfigUtil.getLegacyFactoryConfig(this.getConfigurationAdmin(), this.factoryPid, null, this.configPid);
+                }
```

Comment:
I don't think this is a complete fix (together with https://github.com/apache/sling-org-apache-sling-installer-provider-jcr/pull/9). The `original.pid` property is not leveraged. Installation would need to be skipped if the `original.pid` property is present AND a configuration with that PID exists in config admin.

Reply:
I agree that I haven't checked whether the `original.pid` property is present or not cause I didn't need it (may be we can remove it altogether).

Regarding the checking of config, the `ConfigUtil.getLegacyFactoryConfig` does check it [here](https://github.com/apache/sling-org-apache-sling-installer-factory-configuration/blob/master/src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUtil.java#L251-L256)

Reply:
I think this change here is actually "good enough".

Reply:
@rishabhdaim I believe we need an analogous change in `ConfigRemoveTask`. WDYT?

Reply:
> Regarding the checking of config, the ConfigUtil.getLegacyFactoryConfig does check it [here](https://github.com/apache/sling-org-apache-sling-installer-factory-configuration/blob/master/src/main/java/org/apache/sling/installer/factories/configuration/impl/ConfigUtil.java#L251-L256)
I had another look at this code. I think it doesn't do what we need here. Let me explain with a pseudo test:

```
var factoryPid = "org.apache.sling.FooFactory";
var servicePid = "org.apache.sling.FooFactory~abc";
assertEquals("org.apache.sling.FooFactory.abc", factoryPid + "." + configPid); // fails
```

Reply:
Basically, we want to treat `"org.apache.sling.FooFactory.abc"` as an alias for `"org.apache.sling.FooFactory~abc"`. Then there should also be no issue with double-installation of configurations due to renaming the service-pid.

I wonder if we could (and should?) even drop the change in `o.a.s.installer.provider.jcr` in favour of this implicit aliasing? Configurations would then still be persisted with the dot, but unless already present with the dot, they would be installed with the tilde. I'm not 100% sure that this would work, but it would mean only changing (and deploying) one bundle.

Reply:
> @rishabhdaim I believe we need an analogous change in `ConfigRemoveTask`. WDYT?

If I try to remove the config, it gets deleted from both servers. Not sure, why we need to make changes in ConfigRemoveTask.

Reply:
We need it for deletion in case the config is removed on another cluster node, where it has the '~'-separated format.

Reply:
added in https://github.com/apache/sling-org-apache-sling-installer-factory-configuration/pull/13/commits/fa4ef701943d95402a5e952ffda6338fdbe29f4a